lib/client/api.go

```diff
@@ -3163,7 +3168,12 @@ func (tc *TeleportClient) getSSHLoginFunc(pr *webclient.PingResponse) (SSHLoginF
 			if !pr.Auth.AllowHeadless {
 				return nil, trace.BadParameter("headless disallowed by cluster settings")
 			}
-			// TODO (Joerger): Add headless login flow.
+			if tc.AllowHeadless {
+				return func(ctx context.Context, priv *keys.PrivateKey) (*auth.SSHLoginResponse, error) {
+					return tc.headlessLogin(ctx, priv)
+				}, nil
+			}
+			log.Debug("Headless login is disabled for this command. Only 'tsh ls', 'tsh ssh', and 'tsh scp' are supported. Defaulting to local authentication methods.")
```

Comment:
Should this default to other auth method, or fail if headless auth was requested but not supported?

Reply:
I currently have it set up to default to other auth methods. The primary use case for this is so that you can set `cluster_auth_preference.connector = headless` and it will still default to local connector when headless is not applicable.

```console
< local machine > $ tsh login ... 
# logs in with local authentication method
...

< remote machine > $ tsh ssh ...
# initiates headless
tsh approve request-id
...

< local machine > tsh approve request-id
# Uses local authenticaiton (~/.tsh) to approve
```